### PR TITLE
Refactor network.py to support a larger number of servers

### PR DESCRIPTION
**Pull request by eliran-stratoscale**

### CODE REVIEW

File: rackattack/physical/network.py [hunk 1]
```diff
@@ -1,13 +1,33 @@
...
+    return validRackattackAddresses, subnetMask
+
+_NETMASK_PREFIX_LENGTH = 22
+_NR_RESERVED_HOSTS = 10
+_IP_ADDRESS_FORMAT = "192.168.%d.%d"
+_FIRST_ADDR_IN_SUBNET = _IP_ADDRESS_FORMAT % (0, 0)
+_HOSTS, NETMASK = getHostsInSubnetWithRackattackConvention(_FIRST_ADDR_IN_SUBNET,
+                                                           _NETMASK_PREFIX_LENGTH,
+                                                           _NR_RESERVED_HOSTS)
+GATEWAY_IP_ADDRESS = _IP_ADDRESS_FORMAT % (1, 1)
```
Comment (pavel-stratoscale) [inline rackattack/physical/network.py]: consts in the middle of the file?

File: rackattack/physical/network.py [hunk 2]
```diff
@@ -1,13 +1,33 @@
...
-_IP_ADDRESS_FORMAT = "192.168.1.%d"
-GATEWAY_IP_ADDRESS = _IP_ADDRESS_FORMAT % 1
-BOOTSERVER_IP_ADDRESS = _IP_ADDRESS_FORMAT % 1
-NETMASK = '255.255.255.0'
-_NETWORK_PREFIX = "192.168.1."
-LAST_INDEX = 200
+import ipaddr
+
+
+def getHostsInSubnetWithRackattackConvention(firstAddr, prefixLen, nrReservedAddresses):
```
Comment (pavel-stratoscale) [inline rackattack/physical/network.py]: The name is not that good

File: rackattack/physical/network.py [hunk 3]
```diff
@@ -1,13 +1,33 @@
...
+    subnet = ipaddr.IPv4Network("%(firstAddr)s/%(prefixLen)s" % dict(firstAddr=firstAddr,
+                                                                     prefixLen=prefixLen))
+    subnetMask = str(subnet.netmask)
+    addresses = [str(host) for host in subnet.iterhosts()]
+    validRackattackAddresses = list()
+    for address in addresses:
+        _, _, thirdOctet, fourthOctet = [int(octet) for octet in address.split(".")]
+        if thirdOctet > 1 or (thirdOctet == 1 and fourthOctet > nrReservedAddresses):
+            assert address not in validRackattackAddresses
```
Comment (rotem-stratoscale) [inline rackattack/physical/network.py]: In general it feels that there should be an existing library that already deals with splits, list of unique IPs under certain subnets etc.
Please spend few minutes to check it (for example, something like https://pythonhosted.org/netaddr )

**Comment by eliran-stratoscale** [comment]
Will be rebased, not merged.

**Pull request closed without merging**